default/python/AI/AIDependencies.py

```diff
@@ -782,6 +784,22 @@
     "GRO_ENERGY_META": {FUEL: 2},
 }
 
+DEFAULT_FUEL_EFFICIENCY = 1
+HULL_TAG_EFFECTS = {
```

Comment:
Presumably these tags could also be added to a part, so they don't really need to have "hull" in the names, do they?

Reply:
Mixing here all kinds of tags - I would call that premature optimisation.  Probably the effects for tags on parts would be very different from hull tags. E.g. one could have multiple parts with the same or differing tags. E.g. having GOOD_FUEL_HULL and BAD_FUEL_HULL in the same ship does not make sense.

So i suggest to keep the HULL_TAG_EFFECTS until the need for something more general arises.